--- ipw3945d-1.7.22/ipw3945d-start.net	2006-07-14 20:40:16.000000000 +0200
+++ ipw3945d-1.7.22/ipw3945d-start	2007-03-20 16:10:03.000000000 +0100
@@ -14,6 +14,18 @@
 	return 0
 }
 
+wait_for_net() {
+	# Finally, wait for sysfs entry to show up, timing out after 10 seconds:
+	count=0
+	while [ ! -e /sys/bus/pci/drivers/ipw3945/*/net:* ]; do
+		sleep 0.5
+		count=$((count+1))
+		((count > 20)) && return 1
+	done
+
+	return 0
+}
+
 wait_for_cmd && {
 	# Set ownership of sysfs entry:
 	chown $user /sys/bus/pci/drivers/ipw3945/*/cmd
@@ -27,4 +39,4 @@
 
 	# Launch the regulatory daemon:
 	/sbin/ipw3945d --quiet --pid-file=/var/run/ipw3945d/ipw3945d.pid
-}
+} && wait_for_net
